terraform/shared/ci-mirrors-access.tf

```diff
@@ -0,0 +1,35 @@
+resource "aws_iam_role" "foo" {
+  name = "ci--rust-lang--ci-mirrors"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Principal = {
+          Federated = aws_iam_openid_connect_provider.github_actions.arn
+        }
+        Condition = {
+          StringLike = {
+            "token.actions.githubusercontent.com:sub" = "repo:rust-lang/ci-mirrors:ref:refs/heads/gh-readonly-queue/main/*"
```

Comment:
nit: we might want a different branch name since this does have PutObject permissions?

Reply:
i don't think we can change the branch name, it's what GitHub Merge Queues put code into. I have a different idea for how to organize CI (have the apply be done by a separate job that doesn't build any code, to reduce permissions even more), and there I can just switch to our usual environment-based authorization. I'll do so tomorrow.

Reply:
Switched to an environment, as per https://github.com/rust-lang/ci-mirrors/pull/2